src/sentry/workflow_engine/handlers/condition/slow_condition_query_handlers.py

```diff
@@ -242,3 +260,90 @@ def get_result(model: TSDBModel, group_ids: list[int]) -> dict[int, int]:
             batch_sums.update(get_result(model, issue_ids))
 
         return batch_sums
+
+
+@slow_condition_query_handler_registry.register(Condition.PERCENT_SESSIONS_COUNT)
+@slow_condition_query_handler_registry.register(Condition.PERCENT_SESSIONS_PERCENT)
+class PercentSessionsQueryHandler(BaseEventFrequencyQueryHandler):
+    intervals: ClassVar[dict[str, tuple[str, timedelta]]] = PERCENT_INTERVALS
+
+    def get_session_count(
+        self, project_id: int, environment_id: int | None, start: datetime, end: datetime
+    ) -> int:
+        cache_key = f"r.c.spc:{project_id}-{environment_id}"
+        session_count_last_hour = cache.get(cache_key)
+        if session_count_last_hour is None:
+            with options_override({"consistent": False}):
+                session_count_last_hour = release_health.backend.get_project_sessions_count(
+                    project_id=project_id,
+                    environment_id=environment_id,
+                    rollup=60,
+                    start=end - timedelta(minutes=60),
+                    end=end,
+                )
+
+            cache.set(cache_key, session_count_last_hour, 600)
+        return session_count_last_hour
+
+    def get_session_interval(self, session_count: int, duration: timedelta) -> int | None:
+        if session_count >= MIN_SESSIONS_TO_FIRE:
+            interval_in_minutes = duration.total_seconds() // 60
+            return int(session_count / (60 / interval_in_minutes))
```

Comment:
what does the denominator here mean? You may want to consider putting the "math" in variables (like `interval_in_minutes`) so it's easier to grasp without needing to figure the logic out

Reply:
i'm not entirely sure what this means either, i'm moving it from the existing condition class in `event_frequency.py`
https://github.com/getsentry/sentry/blob/41d1f5e1011d8bf67398aa99fbb5c2bb312df76e/src/sentry/rules/conditions/event_frequency.py#L857-L861
cc @ceorourke ?

Reply:
i found the original PR https://github.com/getsentry/sentry/pull/26021

Reply:
ah gotcha, np if it's a refactor!